src/Entity/OgMembership.php

```diff
@@ -286,9 +298,27 @@ public function setRoles(array $roles = []): OgMembershipInterface {
    * {@inheritdoc}
    */
   public function getRolesIds(): array {
-    return array_map(function (OgRole $role) {
-      return $role->id();
-    }, $this->getRoles());
+    // Only use $this->get() if it is already populated. If it is not available
+    // then use the raw value. This field is not translatable so we do not need
+    // the slow field definition lookup from $this->getTranslatedField().
+    if (isset($this->fields['roles'][LanguageInterface::LANGCODE_DEFAULT])) {
+      $values = $this->get('roles')->getValue();
+    }
+    else {
+      $values = $this->values['roles'][LanguageInterface::LANGCODE_DEFAULT] ?? [];
+    }
+
+    $roles_ids = array_map(function (array $role_data): string {
```

Comment:
$roles_ids = array_column($role_data, 'target_id') does the same

Reply:
```suggestion
    $roles_ids = array_column($role_data, 'target_id');
```